python/tests/integration/arcticdb/version_store/test_snapshot.py

```diff
@@ -439,3 +423,41 @@ def test_remove_from_snapshot_multiple(basic_store_tombstone_and_pruning):
     assert len(versions) == 1
     assert lib.read("s3", as_of="saved").data == 3
     assert lib.read("s2", as_of="saved").data == 2
+
+
+def test_snapshot_tombstoned_key(basic_store_delayed_deletes_v1):
+    lib = basic_store_delayed_deletes_v1
+    symA = "A"
+    symB = "B"
+
+    symA_ver1 = lib.write(symA, 1).version
+    symA_ver2 = lib.write(symA, 2).version
+    with pytest.raises(NoSuchVersionException):
+        lib.read(symA, as_of=symA_ver1)
+
+    lib.write(symB, 3).version
+    symB_ver2 = lib.write(symB, 4).version
+    
+    partial_valid_versions = {symA:symA_ver1, symB:symB_ver2}
+    with pytest.raises(NoSuchVersionException):
+        lib.snapshot("s", versions=partial_valid_versions)
+    assert len(lib.list_snapshots()) == 0
+
+    lib.snapshot("s", versions=partial_valid_versions, allow_partial_snapshot=True)
```

Comment:
I think a separate test for the `allow_partial_snapshot=True` case would be clearer than folding it in to this one

Reply:
Exception `NoSuchVersionException` will be thrown.
I have already broken down the test - check `test_snapshot_not_accept_tombstoned_key`.